File: backend/app/schemas/categoria.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CategoriaBase(BaseModel):
    nombre: str = Field(min_length=1, max_length=100)
    descripcion: str | None = Field(default=None, max_length=255)
    activo: bool = True


class CategoriaCreate(CategoriaBase):
    pass


class CategoriaUpdate(BaseModel):
    nombre: str | None = Field(default=None, min_length=1, max_length=100)
    descripcion: str | None = Field(default=None, max_length=255)
    activo: bool | None = None


class CategoriaOut(CategoriaBase):
    id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/categoria.ts
export interface Categoria {
  id: number;
  nombre: string;
  descripcion: string | null;
  activo: boolean;
  created_at: string;
}

export interface CategoriaCreate {
  nombre: string;
  descripcion?: string | null;
  activo?: boolean;
}

export interface CategoriaUpdate {
  nombre?: string;
  descripcion?: string | null;
  activo?: boolean;
}
